setupGcloudSDK/src/index.ts

```diff
@@ -0,0 +1,228 @@
+import { getReleaseURL } from './format-url';
+import * as downloadUtil from './download-util';
+import * as installUtil from './install-util';
+import { getLatestGcloudSDKVersion } from './version-util';
+
+export { getLatestGcloudSDKVersion };
+
+/**
+ * Checks if gcloud is installed
+ *
+ * @param version (Optional) Cloud SDK version
+ * @return true if gcloud is found in toolpath
+ */
+export function isInstalled(version?: string): boolean {
+  let toolPath;
+  if (version) {
+    toolPath = toolCache.find('gcloud', version);
+    return toolPath != undefined && toolPath !== '';
+  } else {
+    toolPath = toolCache.findAllVersions('gcloud');
+    return toolPath.length > 0;
+  }
+}
+
+/**
+ * Returns the correct gcloud command for OS
+ *
+ * @returns gcloud command
+ */
+export function getToolCommand(): string {
+  // A workaround for https://github.com/actions/toolkit/issues/229
+  // Currently exec on windows runs as cmd shell.
+  let toolCommand = 'gcloud';
+  if (process.platform == 'win32') {
+    toolCommand = 'gcloud.cmd';
+  }
+  return toolCommand;
+}
+
+/**
+ * Checks if the project Id is set in the gcloud config
+ *
+ * @returns true is project Id is set
+ */
+export async function isProjectIdSet(): Promise<boolean> {
+  let output = '';
+  const stdout = (data: Buffer): void => {
+    output += data.toString();
+  };
+  const options = {
+    listeners: {
+      stdout,
+    },
+  };
+
+  const toolCommand = getToolCommand();
+
+  await exec.exec(toolCommand, ['config', 'get-value', 'project'], options);
+  return !output.includes('unset');
+}
+
+/**
+ * Checks if gcloud is authenticated
+ *
+ * @returns true is gcloud is authenticated
+ */
+export async function isAuthenticated(): Promise<boolean> {
+  let output = '';
+  const stdout = (data: Buffer): void => {
+    output += data.toString();
+  };
+  const options = {
+    listeners: {
+      stdout,
+    },
+  };
+
+  const toolCommand = getToolCommand();
+
+  await exec.exec(toolCommand, ['auth', 'list'], options);
+  return !output.includes('No credentialed accounts.');
+}
+
+/**
+ * Installs the gcloud SDK into the actions environment.
+ *
+ * @param version The version being installed.
+ * @param gcloudExtPath The extraction path for the gcloud SDK.
+ * @returns The path of the installed tool.
+ */
+export async function installGcloudSDK(version: string): Promise<string> {
+  // Retreive the release corresponding to the specified version and OS
+  const osPlat = os.platform();
+  const osArch = os.arch();
+  const url = await getReleaseURL(osPlat, osArch, version);
+
+  // Download and extract the release
+  const extPath = await downloadUtil.downloadAndExtractTool(url);
+  if (!extPath) {
+    throw new Error(`Failed to download release, url: ${url}`);
+  }
+
+  // Install the downloaded release into the github action env
+  return await installUtil.installGcloudSDK(version, extPath);
+}
+
+/**
+ * Parses the service account string into JSON
```

Comment:
Missing trailing '.'

Reply:
Here and in a couple other comment headers in this file.

Reply:
done